src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1589,20 +1596,26 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
 
                     var primaryPackages = new List<PackageIdentity> { packageIdentity };
 
-                    var gatherContext = new GatherContext()
-                    {
-                        InstalledPackages = oldListOfInstalledPackages.ToList(),
-                        PrimaryTargets = primaryPackages,
-                        TargetFramework = targetFramework,
-                        PrimarySources = primarySources.ToList(),
-                        AllSources = effectiveSources.ToList(),
-                        PackagesFolderSource = PackagesFolderSourceRepository,
-                        ResolutionContext = resolutionContext,
-                        AllowDowngrades = downgradeAllowed,
-                        ProjectContext = nuGetProjectContext
-                    };
+                    HashSet<SourcePackageDependencyInfo> availablePackageDependencyInfoWithSourceSet = null;
 
-                    var availablePackageDependencyInfoWithSourceSet = await ResolverGather.GatherAsync(gatherContext, token);
+                    using (var sourceCacheContext = new SourceCacheContext())
```

Comment:
This should come from ResolutionContext which is created at a higher level

Reply:
I thought so as well, but then there will be more changes besides we already have `SourceCacheContext` being created in `NuGetPackageManager`/. But I'll see if I can plum it through all the way.